packages/xsnap/test/test-xs-perf.js

```diff
@@ -179,3 +184,62 @@ test('metering switch - start compartment only', async t => {
   await vat.close();
   t.deepEqual(['no meteringSwitch in Compartment'], opts.messages);
 });
+
+/** @param {number} logn */
+function dataStructurePerformance(logn) {
+  // eslint-disable-next-line no-bitwise
+  const n = 1 << logn;
+  // @ts-ignore
+  // eslint-disable-next-line no-undef
+  const send = it => issueCommand(ArrayBuffer.fromString(JSON.stringify(it)));
+  const t0 = performance.now();
+  for (let i = 0; i < 256; i += 1) {
+    const a = [];
+    for (let j = 0; j < n; j += 1) {
+      a.push(j);
+    }
+    const m = new Map();
+    for (let j = 0; j < n; j += 1) {
+      m.set(j, j);
+    }
+    for (let j = 0; j < n; j += 1) {
+      m.get(j);
+    }
+    const s = new Set();
+    for (let j = 0; j < n; j += 1) {
+      s.add(j);
+    }
+    for (let j = 0; j < n; j += 1) {
+      s.has(j);
+    }
+  }
+  const t1 = performance.now();
+  const dur = t1 - t0;
+  // O(n log(n))
+  const rate = (n * logn) / dur;
+  send({ size: n, dur, rate });
+}
+
+test('Array, Map, Set growth is O(log(n))', async t => {
+  const opts = options(io);
+  const vat = xsnap({ ...opts, meteringLimit: 0 });
+  await vat.evaluate(
+    `globalThis.dataStructurePerformance = (${dataStructurePerformance})`,
+  );
+
+  const run = async size => {
+    const {
+      meterUsage: { compute },
+    } = await vat.evaluate(`dataStructurePerformance(${size})`);
+    const r = JSON.parse(opts.messages.pop());
+    t.log({ compute, r });
+    return { compute, r };
+  };
+
+  const { r: r1 } = await run(8);
+  const { r: r2 } = await run(10);
+  const { r: r3 } = await run(12);
+  t.log({ r2_1: r2.rate / r1.rate, r3_2: r3.rate / r2.rate });
+  t.true(r2.rate / r1.rate >= 1);
+  t.true(r3.rate / r2.rate >= 1);
```

Comment:
What do these assertions verify?

Reply:
that the data structure manipulation rate (computed above as `n * logn / delta_t`) stays comparable as the (binary) order of magnitude grows.

It used to be `>= 0.85`. Now that it's `>= 1`, it could just be `r2.rate >= r1.rate`, I suppose.